library/Freckle/App/Kafka.hs

```diff
@@ -0,0 +1,132 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Freckle.App.Kafka
+  ( BrokerAddress(..)
+  , HasKafkaBrokerAddress (..)
+  , envKafkaBrokerAddress
+
+  , KafkaProducer (..)
+  , HasKafkaProducer (..)
+  , kafkaProducer
+
+  , produceAsync
+
+  ) where
+
+import Freckle.App.Prelude
+
+import Blammo.Logging
+import Control.Lens (Lens', view)
+import Data.Aeson (ToJSON, encode)
+import Data.ByteString (toStrict)
+import qualified Data.List.NonEmpty as NE
+import Data.Pool (Pool)
+import qualified Data.Pool as Pool
+import qualified Data.Text as T
+import qualified Freckle.App.Env as Env
+import qualified Kafka.Producer as Kafka
+import UnliftIO.Async (async)
+import UnliftIO.Exception (throwString)
+import Yesod.Core.Lens
+import Yesod.Core.Types (HandlerData)
+
+newtype BrokerAddress = BrokerAddress { getBrokerAddress :: Maybe (NonEmpty Kafka.BrokerAddress)}
+
+class HasKafkaBrokerAddress env where
+  kafkaBrokerAddressL :: Lens' env BrokerAddress
+
+instance HasKafkaBrokerAddress site => HasKafkaBrokerAddress (HandlerData child site) where
+  kafkaBrokerAddressL = envL . siteL . kafkaBrokerAddressL
+
+envKafkaBrokerAddress
+  :: Env.Parser Env.Error BrokerAddress
+envKafkaBrokerAddress = BrokerAddress <$> addressOrNothing
+
+ where
+  addressOrNothing = (parseKafkaBrokerAddress <$> parseKey) <|> pure Nothing
+  parseKey = Env.var Env.nonempty "KAFKA_BROKER_ADDRESS" mempty
+  parseKafkaBrokerAddress = NE.nonEmpty . fmap Kafka.BrokerAddress . T.splitOn ","
+
+data KafkaProducer
+  = NullKafkaProducer
+  | KafkaProducerPool (Pool Kafka.KafkaProducer)
+
+class HasKafkaProducer env where
+  kafkaProducerL :: Lens' env KafkaProducer
+
+instance HasKafkaProducer site => HasKafkaProducer (HandlerData child site) where
+  kafkaProducerL = envL . siteL . kafkaProducerL
+
+kafkaProducer
+  :: ( MonadReader env m
+     , HasKafkaBrokerAddress env
+     , MonadUnliftIO m
+     )
+  => Int
+  -- ^ The number of stripes (distinct sub-pools) to maintain.
```

Comment:
The docs here were stolen wholesale from the underlying `Pool.createPool`. 

I thought about giving some sensible defaults instead of parametrizing this function, but this flexibility for the client seemed nice.